### PR TITLE
Use edit role for Che SA

### PR DESCRIPTION
**Pull request by ghost**
### What does this PR do?

It looks like Che service account may have edit role to be able to create workspaces in the same namespace with Che server pod

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/11142

### CODE REVIEW

File: deploy/openshift/templates/che-server-template.yaml [hunk 1]
```diff
@@ -24,7 +24,7 @@ objects:
   metadata:
     name: che
   roleRef:
-    name: admin
+    name: edit
```
Comment (sleshchenko) [inline deploy/openshift/templates/che-server-template.yaml:27]: maybe it makes describe in the documentation that CHE_INFRA_OPENSHIFT_PROJECT must be set to:
1. the same with Che Server namespace
2. empty value then oauth provider must be used
3. any value different from Che Server namespace then cluster admin roles should be bound to Che service account (not sure that we need this way of configuration)

**Review by sleshchenko — APPROVED** [review]
LGTM

Please change PR title to make clear that it affects OpenShift deployment only

File: deploy/openshift/templates/che-server-template.yaml (hunk 1 above)
Comment (ghost) [reply deploy/openshift/templates/che-server-template.yaml:27]: Currently Che service account does not have cluster wide privileges so it is a general documentation improvement not really related to this particular PR. But good point.

File: deploy/openshift/templates/che-server-template.yaml [hunk 2]
```diff
@@ -24,7 +24,7 @@ objects:
   metadata:
     name: che
   roleRef:
-    name: admin
```
[the remark this answers is not in the document]
Comment (ghost) [reply deploy/openshift/templates/che-server-template.yaml:27]: In this case, a cluster admin should grant Che SA cluster-admin privileges:

```
oc adm policy add-cluster-role-to-user self-provisioner system:serviceaccount:eclipse-che:che
```
For k8s we do that out of the box: https://github.com/eclipse/che/blob/master/deploy/kubernetes/helm/che/templates/cluster-role-binding.yaml#L16

And this makes it possible to set [this env]( https://github.com/eclipse/che/blob/master/deploy/kubernetes/helm/che/templates/configmap.yaml#L51) to [empty string](https://github.com/eclipse/che/blob/master/deploy/kubernetes/helm/che/values.yaml#L47) by default and things work out of the box.

So, to sum it up - on OpenShift we NEVER granted Che SA cluster-admin role while it was true for Kubernetes Helm charts. Hence, if we do the same change for k8s, then, https://github.com/eclipse/che/blob/master/deploy/kubernetes/helm/che/values.yaml#L47 should be namespace where Che SA has been created. And if a cluster admin wants Che to create workspaces in individual namespaces, Che SA is granted a cluster-admin role and https://github.com/eclipse/che/blob/master/deploy/kubernetes/helm/che/values.yaml#L47 becomes an empty string.

[the remark this answers is not in the document]
Comment (garagatyi) [reply deploy/openshift/templates/che-server-template.yaml:27]: Does it mean that we don't have right now an OS deployment option where Che creates namespaces?

**Review by garagatyi — APPROVED** [review]

**Comment by eivantsov** [comment]
ci-test

File: deploy/openshift/templates/che-server-template.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (l0rd) [reply deploy/openshift/templates/che-server-template.yaml:27]: @garagatyi this option should be supported when Che is [configured to use users tokens](http://www.eclipse.org/che/docs/openshift-admin-guide.html). Giving cluster-admin privileges to Che is something that we should not support: you can do it at your own risk :-)

**Review by l0rd — APPROVED** [review]

**Comment by riuvshin** [comment]
ci-test build report:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/779/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/779//Selenium_20tests_20report/)
[selenium tests report data](http://tests.codenvycorp.com/selenium_test_report_data/11177)
docker image: eclipseche/che-server:11177
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.
